### DISCUSSION TITLE
how to compile project with new plugin?

### DISCUSSION THREAD

**Discussion by cameronmgj** [pear-devs/pear-desktop#1007] [Q&A] [answered]
I've tried installing yarn, electron builder, and node but can't figure out what I'm supposed to do. Yarn doesn't seem to have an interface at all let alone a cli to write in, and the other two come up with errors in their cli's

how are you supposed to run yarn? I'm probably just being stupid tbh, but I'm new to github and wanted to try making a plugin; I have pretty much no idea what I'm doing except for the actual code itself :p

**Comment by Zo-Bro-23** [comment, marked as answer]
You say you installed Node, Yarn, and Electron Builder, but all three have different functions and you need to install all three.

There is a build section in the readme for how to build the app, and the method is the same after adding plugins too. I'll repeat the steps here:
- Clone the repo or download the ZIP file
- Run `yarn` in the command line
- Run `yarn build:OS`, where OS can be `win`, `linux`, or `mac`

Note that NodeJS and Yarn both need to be installed (and added to path) for this to work, but Electron Builder need not be installed since running `yarn` installs it automatically. Also, the repo only needs to be cloned once and same for the `yarn` command. Just run `yarn build` after making the necessary changes for adding the plugin. Hope this helped!

**Reply by cameronmgj** [reply]
thanks this made it a little clearer, after playing around i managed to get it working :) my plugin doesn't work but that's probably just my code XD

**Comment by da-anda** [comment]
does anybody have any instructions on how to build it on a Windows host or inside a docker container? My attempts failed so far, as lots and lots of dependencies/libs are missing (libdrm, gtk, gbm, ...). I suppose it only works on a linux env with a running gbm/x11 session?

edit: or is there any way to mess around with custom plugins (by using a custom build env that is only compiling the TS/JS but ignores any electron stuff) in a downloaded version?

**Reply by JellyBrick** [reply]
`npm i` -> `npm dist:win:x64`
